lib/app/service/messenger/widget/messenger_service_widget.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:app/app/service/messenger/cubit/messenger_cubit.dart';
+import 'package:app/app/service/service.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:ui_components_lib/ui_components_lib.dart';
+
+class MessengerServiceWidget extends StatefulWidget {
+  const MessengerServiceWidget({
+    required this.child,
+    super.key,
+  });
+
+  final Widget child;
+
+  @override
+  State<MessengerServiceWidget> createState() => _MessengerServiceWidgetState();
+}
+
+class _MessengerServiceWidgetState extends State<MessengerServiceWidget> {
+  final MessengerCubit _cubit = MessengerCubit();
+  bool _isSnackbarShown = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _cubit.init();
+  }
+
+  void _onDismiss() {
+    _isSnackbarShown = false;
+    _showNextMessage();
+  }
+
+  void _showNextMessage() {
+    // Wait for the current snackbar to be dismissed
+    if (_isSnackbarShown) {
+      return;
+    }
+
+    // We can't use current context here because it's not contains Navigator
+    final ctx = NavigationService.navigatorKey.currentState?.context;
```

Comment:
Why like static method? It's in `GetIt`

Reply:
Let's just leave it as it is. Temporary :)